cli/src/main.rs

```diff
@@ -56,20 +60,11 @@ async fn display_users(pool: &PgPool) -> color_eyre::Result<()> {
     Ok(())
 }
 
+
 async fn sign_up(pool: &PgPool, username_opt: Option<String>) -> color_eyre::Result<()> {
     if Path::new("auth.txt").exists() {
-        let mut file = fs::File::open("auth.txt")?;
-        let mut contents = String::new();
-        file.read_to_string(&mut contents)?;
-        let user_id = Uuid::parse_str(contents.trim())?;
-
-        match db::get_username_by_id(pool, user_id).await? {
-            Some(existing_username) => {
-                println!("User ID already registered with username: {existing_username}");
-                return Ok(());
-            }
-            None => println!("No existing user found with this ID, proceeding with signup."),
-        }
+        check_status(pool).await?;
+        return Ok(());
```

Comment:
To avoid repeating the same logic, I called the function `check_status` here

Reply:
Love that!